manifests/pe_metric.pp

```diff
@@ -20,7 +20,28 @@
       'absent'  => absent,
     },
   }
-
+  ##Ref SUP-346
+  $additional_metrics = [
+    { 'name' => 'compiler.find_node',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.compiler.find_node" },
+    { 'name' => 'puppetdb.query',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.query" },
+    { 'name' => 'puppetdb.resource.search',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.resource.search" },
+    { 'name' => 'puppetdb.facts.encode',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.facts.encode" },
+    { 'name' => 'puppetdb.command.submit.replace facts',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.command.submit.replace facts" },
+    { 'name' => 'puppetdb.catalog.munge',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.catalog.munge" },
+    { 'name' => 'puppetdb.command.submit.replace catalog',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.command.submit.replace catalog" },
+    { 'name' => 'puppetdb.report.convert_to_wire_format_hash',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.report.convert_to_wire_format_hash" },
+    { 'name' => 'puppetdb.command.submit.store report',
+      'url'  => "puppetserver:name=puppetlabs.${::hostname}.puppetdb.command.submit.store report" },
+  ]
```

Comment:
Since these metrics are specific to Puppet Server, the should go over to `puppetserver.pp`. The `pe_metric.pp` type is shared by all services, so having them here will apply the collection to PuppetDB and Orchestrator, where these measurements are not present.

The metrics should also be collected only if `$::pe_server_version` is less than 2018.1.0 as we've added these to the core set reported by the `/status` endpoint.